internal/resolver/delegatingresolver/delegatingresolver_test.go

```diff
@@ -0,0 +1,71 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package delegatingresolver
+
+import (
+	"testing"
+
+	"google.golang.org/grpc/internal/testutils"
+	"google.golang.org/grpc/resolver"
+	"google.golang.org/grpc/resolver/manual"
+	"google.golang.org/grpc/serviceconfig"
+)
+
+// TestDelegatingResolverNoProxy verifies the behavior of the delegating resolver when no proxy is configured.
+func TestDelegatingResolverNoProxy(t *testing.T) {
+	t.Setenv("HTTPS_PROXY", "")               // Explicitly set proxy environment to empty to mimic no proxy environment set
```

Comment:
Again, these should set variables exported by the envconfig package. Also, they should revert those variables to their original values at the end of the test.